File: study/TimeStamp/app/src/main/java/com/example/timestamp/MainActivity.java
package com.example.timestamp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import java.text.SimpleDateFormat;
import java.util.Date;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        /**
         * 方式1
         * 注:默认使用的是北京时区,无论是UTC时间戳还是北京时间戳,格式化后都是北京时间。
         */
        Long unixtimestamp = 1641537092L;
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        String format1 = simpleDateFormat.format(new Date(unixtimestamp * 1000));
        System.out.println(format1);//2022-01-07 14:31:32

    }
}

File: AutomatedTesting/app/src/main/java/com/example/automatedtesting/MainActivity.kt
package com.example.automatedtesting

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.util.Patterns
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import com.example.automatedtesting.R


class MainActivity : AppCompatActivity() {

    var edtUserName : EditText? = null

    var edtPassWord : EditText? = null

    var btnLogin : Button? = null

    var loginLogic : LoginLogic = LoginLogic()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        edtUserName = findViewById(R.id.edtUsename)
        edtPassWord = findViewById(R.id.edtPassword)
        btnLogin = findViewById(R.id.btnLogin)
        btnLogin?.setOnClickListener {
            var success = loginLogic.login(this,edtUserName?.text.toString(),edtPassWord?.text.toString())
            if (success){
                var intent = Intent(this,LoginActivity::class.java)
                startActivity(intent)
            }
        }
    }


    class LoginLogic {
        fun login(context: Context?, username: String, password: String): Boolean {
            return if (!isUserNameValid(username) || !isPasswordValid(password)) {
                false
            } else {
                //通过服务器判断账户及密码的有效性
                val result = checkFromServer(username, password)
                if (result) {
                    //登录成功保持本地的信息
                    Log.e("TAG","登录成功保持本地的信息")
                   // SharedPreferencesUtils.put(context, username, password)
                }
                result
            }
        }

        private fun isUserNameValid(username: String?): Boolean {
            if (username == null) {
                return false
            }
            return if (username.contains("@")) {
                Patterns.EMAIL_ADDRESS.matcher(username).matches()
            } else {
                !username.trim { it <= ' ' }.isEmpty()
            }
        }

        fun isPasswordValid(password: String?): Boolean {
            return password != null && password.trim { it <= ' ' }.length > 5
        }

        companion object {
            // 为了进行演示，去除通过服务器鉴定的逻辑，当用户输入特定账号及密码为时则验证成功
            private fun checkFromServer(username: String, password: String): Boolean {
                return username == "123@163.com" && password == "123456"
            }
        }
    }
}
